### PR TITLE
generalized fake fishes

### PR DESCRIPTION
**Pull request by janscience**
I generalized Juans generate_wavefish() and generate_pulsefish()  functions, added a few specialized functions for specific fishes and extended the docu.

I also simplified the plotting in fakefish main. We should keep the example plots in the main functions as simple as possible. No special colors, alphas, fontsizes, etc. These are examples that other people might want to use to adapt them to their own purposes. They might not care about our lates fancy plot styles.

### CODE REVIEW

File: thunderfish/fakefish.py [hunk 1]
```diff
@@ -1,123 +1,233 @@
...
-    data = np.sin(2 * np.pi * time * freq)
-    for har in range(1, harmonics):
-        data += (np.sin(2*np.pi*time*(har*freq))) * 0.1*har
-
-    # Insert some noise
+    time = np.arange(0, duration, 1./samplerate)
+    data = np.zeros(len(time))
+    if np.isscalar(amplitudes):
+        amplitudes = [amplitudes]
+    for har, ampl in enumerate(amplitudes):
```
Comment (jfsehuanes) [inline thunderfish/fakefish.py:51]: I don't understand what you mean with amplitudes... but perhaps a live explanation is more successful.

File: thunderfish/fakefish.py [hunk 2]
```diff
@@ -1,123 +1,233 @@
...
-    :param pk_std: (float). std of the positive part of the pulse in sec.
-    :param pk_amplitude: (float). Factor for regulating the positive part of the pulse.
-    :param tr_std: (float). std of the negative part of the pulse in sec.
-    :param tr_amplitude: (float). Factor for regulating the negative part of the pulse.
-    :param tr_time:
-
-    :return: (array). Data with pulses at the given frequency.
+    :param duration: (float). Duration of the generated data in seconds.
+    :param noise_std: (float). Standard deviation of additive Gaussian white noise.
+    :param jitter_cv: (float). Gaussian distributed jitter of pulse times as coefficient of variation of inter-pulse intervals.
```
Comment (jfsehuanes) [inline thunderfish/fakefish.py:83]: nice feature!

File: thunderfish/fakefish.py [hunk 3]
```diff
@@ -1,123 +1,233 @@
...
+    x = np.arange(-4.*peak_stds[min_time_inx] + peak_times[min_time_inx],
+                  4.*peak_stds[max_time_inx] + peak_times[max_time_inx], 1.0/samplerate)
+    pulse_duration = x[-1] - x[0]
+    
+    # generate a single pulse:
+    pulse = np.zeros(len(x))
+    for time, ampl, std in zip(peak_times, peak_amplitudes, peak_stds):
+        pulse += ampl * np.exp(-0.5*((x-time)/std)**2) 
+
+    # paste the pulse onto the noise floor:
```
Comment (jfsehuanes) [inline thunderfish/fakefish.py:113]: typo: "into"

File: thunderfish/fakefish.py [hunk 4]
```diff
@@ -1,123 +1,233 @@
...
-    pulse = np.exp(-0.5 * (x/pk_std)**2) * pk_amplitude  # Peak Gaussian
-    pulse -= np.exp(-0.5 * ((x-tr_time)/tr_std)**2) * tr_amplitude  # Trough Gaussian
+    return data
 
-    # Now we need to set the pulse into some baseline with noise.
-    time = np.arange(0, time_len, 1. / samplerate)
 
-    data = np.random.randn(len(time)) * noise_fac
+def generate_monophasic_pulses(frequency=100.0, samplerate=44100., duration=1.,
+                               noise_std=0.01, jitter_cv=0.1):
```
Comment (jfsehuanes) [inline thunderfish/fakefish.py]: so if I understood correctly, you just vary the noise_std and by that you get either a monophasic or a biphasic pulse? that's nice.

File: thunderfish/fishfinder.py [hunk 5]
```diff
@@ -1,878 +1,878 @@
-#!/usr/bin/python
```
Comment (jfsehuanes) [inline thunderfish/fishfinder.py:1]: Could you briefly summarize what you changed in the fishfinder?  Or just simply ignore changes? :)

**Comment by jfsehuanes** [comment]
Nice updates on the module!

File: thunderfish/fishfinder.py (hunk 5 above)
Comment (janscience) [inline thunderfish/fishfinder.py:1]: I did not change anything, but github marks it as completely changed. fishfinder.py is still working. I consider this a bug and would just ignore it.

**Pull request merged**